core/domain/user_id_migration.py

```diff
@@ -499,15 +478,22 @@ def map(model):
                 yield ('FAILURE - %s' % model_class.__name__, model.id)
         elif (model_class.get_user_id_migration_policy() ==
               base_models.USER_ID_MIGRATION_POLICY.COPY_AND_UPDATE_ONE_FIELD):
-            if (ModelsUserIdsHaveUserSettingsVerificationJob
-                    ._check_id_and_user_id_exist(model.id, model.user_id)):
+            user_id = model.user_id
+            if user_id is None:
+                yield ('SUCCESS_NONE - %s' % model_class.__name__, model.id)
+            elif (ModelsUserIdsHaveUserSettingsVerificationJob
+                    ._check_id_and_user_id_exist(model.id, user_id)):
                 yield ('SUCCESS - %s' % model_class.__name__, model.id)
             else:
                 yield ('FAILURE - %s' % model_class.__name__, model.id)
         elif (model_class.get_user_id_migration_policy() ==
               base_models.USER_ID_MIGRATION_POLICY.ONE_FIELD):
-            if (ModelsUserIdsHaveUserSettingsVerificationJob
-                    ._check_one_field_exists(model)):
+            user_id = model.to_dict()[
```

Comment:
Note -- there should be a comment to explain why access to a protected member is necessary in this case and why it can't be done any other way.

Reply:
Added.